src/main/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/OrganisationService.java

```diff
@@ -63,48 +67,59 @@ public OrganisationService(PrdService prdService,
 
-        int toleranceSeconds = Integer.parseInt(tolerance);
-        LocalDateTime sinceTime = orgLastBatchRunTime.minusSeconds(toleranceSeconds);
-        String formattedSince = ISO_DATE_TIME_FORMATTER.format(sinceTime);
+            int toleranceSeconds = Integer.parseInt(tolerance);
+            LocalDateTime sinceTime = orgLastBatchRunTime.minusSeconds(toleranceSeconds);
+            String formattedSince = ISO_DATE_TIME_FORMATTER.format(sinceTime);
 
-        Integer accessTypeMinVersion = accessTypesEntity.getVersion().intValue();
-        OrganisationsResponse organisationsResponse = prdService
-                .retrieveOrganisations(formattedSince, 1, Integer.valueOf(pageSize)).getBody();
-        writeAllToOrganisationRefreshQueue(organisationsResponse, accessTypeMinVersion);
+            Integer accessTypeMinVersion = accessTypesEntity.getVersion().intValue();
+            OrganisationsResponse organisationsResponse = prdService
+                    .retrieveOrganisations(formattedSince, 1, Integer.valueOf(pageSize)).getBody();
+            writeAllToOrganisationRefreshQueue(organisationsResponse, accessTypeMinVersion);
 
-        int page = 2;
-        boolean moreAvailable = organisationsResponse.getMoreAvailable();
-        while (moreAvailable) {
+            int page = 2;
+            boolean moreAvailable = organisationsResponse.getMoreAvailable();
+            while (moreAvailable) {
 
-            organisationsResponse = prdService
-                    .retrieveOrganisations(formattedSince, page, Integer.valueOf(pageSize)).getBody();
-            writeAllToOrganisationRefreshQueue(organisationsResponse, accessTypeMinVersion);
-            moreAvailable = organisationsResponse.getMoreAvailable();
-            page++;
+                organisationsResponse = prdService
+                        .retrieveOrganisations(formattedSince, page, Integer.valueOf(pageSize)).getBody();
+                writeAllToOrganisationRefreshQueue(organisationsResponse, accessTypeMinVersion);
+                moreAvailable = organisationsResponse.getMoreAvailable();
+                page++;
+            }
+            batchLastRunTimestampEntity.setLastOrganisationRunDatetime(LocalDateTime
+                    .ofInstant(batchRunStartTime.getDate(), ZoneOffset.systemDefault()));
+            batchLastRunTimestampRepository.save(batchLastRunTimestampEntity);
+        } catch (Exception exception) {
+            processMonitorDto.markAsFailed(exception.getMessage());
+            processEventTracker.trackEventCompleted(processMonitorDto);
+            throw exception;
         }
-        batchLastRunTimestampEntity.setLastOrganisationRunDatetime(LocalDateTime.ofInstant(batchRunStartTime.getDate(),
-                ZoneOffset.systemDefault()));
-        batchLastRunTimestampRepository.save(batchLastRunTimestampEntity);
+        processEventTracker.trackEventCompleted(processMonitorDto);
```

Comment:
you need to mark as successful when there isn't an exception before you mark event as completed on 122